runtime/java/src/common/CopperUtil.java

```diff
@@ -0,0 +1,360 @@
+package common;
+
+import common.exceptions.SilverInternalError;
+import common.javainterop.ConsCellCollection;
+import edu.umn.cs.melt.copper.compiletime.dumpers.XMLSpecDumper;
+import edu.umn.cs.melt.copper.compiletime.logging.CompilerLogger;
+import edu.umn.cs.melt.copper.compiletime.pipeline.AuxiliaryMethods;
+import edu.umn.cs.melt.copper.compiletime.spec.grammarbeans.*;
+import edu.umn.cs.melt.copper.compiletime.spec.grammarbeans.visitors.ParserSpecProcessor;
+import edu.umn.cs.melt.copper.main.CopperDumpType;
+import edu.umn.cs.melt.copper.main.CopperIOType;
+import edu.umn.cs.melt.copper.main.CopperPipelineType;
+import edu.umn.cs.melt.copper.main.ParserCompiler;
+import edu.umn.cs.melt.copper.main.ParserCompilerParameters;
+import edu.umn.cs.melt.copper.runtime.engines.semantics.VirtualLocation;
+import edu.umn.cs.melt.copper.runtime.io.Location;
+import edu.umn.cs.melt.copper.runtime.logging.CopperException;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import silver.core.Init;
+import silver.core.NIOVal;
+import silver.core.NLocation;
+
+public final class CopperUtil {
+  private static Location svToCuLocation(NLocation locTerm) {
```

Comment:
The locations passed in here are just the name of the file and don't include the grammar name, which isn't super helpful in error messages when there are multiple files with the same name.  Idea: pass in the grammar name here too (which could be its own annotation on the `Syntax` nonterminals?)

Reply:
Are you suggesting, like, putting it in instead of / in addition to the filename? Don't we already know the grammar from looking at the filename?

Reply:
The file name in a location is just the short name of the file that was parsed.  Like consider maybe prepending the grammar name to the file name in the location.

Reply:
ah, okay, didn't realize that. will do!

Reply:
Fixed in e7d9f87f6db5718f4bce2a8c795804a8f9504aee, 24ebe575cdd67538e4630cf4b99d8976e4e6f3ec, and 6508ee91bf13c8beb3e63bad73bf16251226bea4.